programming/sr/radio/index.md

```diff
@@ -0,0 +1,82 @@
+---
+layout: page
+title: Radio
+---
+
+Radio
+=====
+
+<div class="info">
+This documentation refers to a feature which is only available within the simulator.
+</div>
+
+The `sr.robot` library contains support for detecting radio transmitters with the simulated radio unit on the robot.
+Radio transmitters are attached to various items in the Student Robotics arena.
+Each transmitter encodes their identity in a machine-readable way, which means that robots can identify these objects.
+
+Using the signal strength and bearing of the received radio signals, you are able to
+determine the distance and direction of a transmitter in 3D space relative to
+the radio. Therefore, if the robot can detect transmitters that is at a fixed
+location in the arena, a robot can calculate its exact position in the arena.
+
+The `sr.robot` library provides this through a `radio` attached to your `Robot`,
+which is can perform a `sweep` to detect transmitters:
+
+~~~~~ python
+from sr.robot import *
+R = Robot()
+transmitters = R.radio.sweep()
+~~~~~
+
+When called, the `sweep` function uses the radio reciever to scan for nearby transmitters.
+It returns a list of `Target` objects, each of which describes one of the transmitters that were found within range.
+A detailed description of the attributes of `Target` objects is provided [later in this page](#Target).
+
+Here's an example that will repeatedly print out the bearing and signal stength of each arena transmitter in range:
+
+~~~~~ python
+from sr.robot import *
+R = Robot()
+
+while True:
+    transmitters = R.radio.sweep()
+    print("I found", len(transmitters), "transmitter(s):")
+
+    for tx in transmitters:
+        print(" - Transmitter {0} Bearing: {1} with a signal strength of {2}".format(
+            tx.target_info.station_code,
+            tx.bearing,
+            tx.signal_strength,
+        ))
+~~~~~
+
+
+[Objects of the Radio System](#radio_objects) {#radio_objects}
+===================================
+
+[`Target`](#Target) {#Target}
+----------
+
+A `Target` object contains information about a _detected_ transmitter.
+It has the following attributes:
+
+target_info
+: A [`TargetInfo`](#TargetInfo) object containing information about the transmitter that was detected.
+
+signal_strength
+: The measured strength of the signal as a float.
+
+bearing
+: A float giving the angle to the `Target` in radians.
```

Comment:
`bearing` is normally a compass term, should we expose it in degrees instead?

Which direction is positive and where is zero? (If it were in degrees it would be obvious...)

Reply:
f05dc7418d2c3da1555d97abf023aebcb755fb80